### ISSUE TITLE
[PROPOSAL] Clean up editor toolbar tools via Settings tool

### ISSUE THREAD

**Issue by twMat** [TiddlyWiki/TiddlyWiki5#6515]
**Problem**

This default editor toolbar is *intimidating*. It is just *too many buttons*!

<img width="565" alt="image" src="https://user-images.githubusercontent.com/4534056/157193091-c6a1cffb-507c-480e-9541-d84a7c66a92b.png">

**Proposal**

* replace the editor toolbar "more button" <img width="25" alt="image" src="https://user-images.githubusercontent.com/4534056/157194515-1789875d-c3cb-4665-b51f-e244306ff772.png">
* ...with a new editor toolbar *settings* button <img width="25" alt="image" src="https://user-images.githubusercontent.com/4534056/157200388-18b30492-fd94-4c9a-9af8-a5cedd9a7ad3.png">
* Clicking it would show a dropdown with alternatives for *all* the items (relevant for the specific editor type)<br>...and with **checkboxes** to pin/unpin to the toolbar<br>...and the titles would actually be **buttons**, like in the *sidebar > Tools* list.

Thus the user can either click the button to *apply* the command, or choose to pin/unpin it to the visible tools!

**Reflections**

IMO the *default* UI for this particular area should appeal to beginners. This can be contrasted to e.g something deep inside the Controlpanel that does not *immediately* face even someone just evaluating TW. Still, the proposed functionality makes it trivial to pin more stuff to the toolbar. First impressions are sensitive, so the default tools should probably be similar to what we see in other services with editors, perhaps:

<img width="563" alt="image" src="https://user-images.githubusercontent.com/4534056/157200495-06637c25-5de7-4add-9d46-1cf24af8f6b6.png">

Anyone interested in this matter might also find this discussion/proposal on a [simplified edit view layout](https://github.com/Jermolene/TiddlyWiki5/discussions/6416) interesting.

**Comment by pmario** [comment]
I'm 100% with you. ... also the linked **Proposal for new edittemplate layout** #6416 makes a lot of sense. 
I would add H1 and H2 then I would be OK ...

The only problem is, that we need a mechanism, that allows us to use toolbar shortcuts, even if the toolbar button is hidden. 
That's the main reason, why my toolbar looks more like this, because I need the shortcuts.

![grafik](https://user-images.githubusercontent.com/374655/157241784-3fd3d2cf-8907-4a8d-88a2-e7bb03786171.png)

**Comment by CodaCodr** [comment]
> we need a mechanism, that allows us to use toolbar shortcuts, even if the toolbar button is hidden.

Like make the associated action a "command" which is targeted by both the shortcut and the button.

**Comment by twMat** [comment]
@pmario wrote

>we need a mechanism, that allows us to use toolbar shortcuts, even if the toolbar button is hidden.

Do you mean *keyboard* shortcuts? I'm not experienced with such, but can a keyboard shortcut not be used if the tool button is hidden? If this is not the case, maybe a keyboard shortcut could be made so that it (visibly or invisibly) opens the hiding place to enable the triggering shortcut? (Maybe this is what @CodaCodr means?)

But, regardless, does this make the OP a no-go until a shortcut issue is resolved? That would be a pity.

>If there would be H1 and H2 then I would be OK ...

LOL! If you help to implement this, I'd even vote for a button with a hardcoded link to wikilabs! As long as one can use shortcuts for it, of course.

**Comment by pmario** [comment]
>Not that it addresses your issue here, but the $keyboard widget handles ((shortcuts)), too.

Yes ... now. ... It didn't work that way as the toolbar buttons where implemented. .. We probably have much better possibilities now, than we had as they where invented.
